kernel/thread.c

```diff
@@ -348,40 +425,173 @@ long myst_syscall_wait4(
     for (;;)
     {
         /* search the zombie list for a process thread */
-        for (myst_thread_t* p = _zombies_head; p; p = p->znext)
+        for (p = _zombies_head; p; p = p->znext)
         {
-            bool match = false;
-
             if (!myst_is_process_thread(p))
```

Comment:
Every thread in the zombie list is a process thread, right? Do we still need the check?